test/test_controller.py

```diff
@@ -1282,3 +1283,198 @@ def test_collect_binlogs_to_purge():
     assert binlogs_to_purge == binlogs
     assert only_inapplicable_binlogs is False
     log.info.assert_any_call("Binlog %s has been replicated to all servers, purging", 3)
+
+
+@patch.object(RestoreCoordinator, "MAX_BASEBACKUP_ERRORS", 2)
+def test_restore_failed_basebackup(
```

Comment:
I'm a great nitpicker for naming; I don't understand what this should test.

test_basebackup_is_marked_as_failed_if_restore_fails ?

Reply:
I wouldn't say the test was intended only to check if the basebackup was marked. I realized that the restore/fail  flow was not entirely tested. So, the main idea is to verify if the selected backup fails, the controller is be capable of retrying with previous backup. Making sure the failed backup is been marked is just one part of it. 
 So that's why I went with a more "generalized" naming, do you have a better suggestion?

Reply:
My suggestion would be to split tests (one behaviour asserted per test) and use names that assert the behaviour you'd like to test. I know that not all of our tests follow that convention, but it's just great for self-documentation and for people reading it in the future.